packages/core/src/agent/MessageReceiver.ts

```diff
@@ -209,7 +210,7 @@ export class MessageReceiver {
     }
 
     // Cast the plain JSON object to specific instance of Message extended from AgentMessage
-    return JsonTransformer.fromJSON(message, MessageClass)
+    return JsonTransformer.fromJSON(message, MessageClass, { validate: false })
```

Comment:
Do you get any errors when calling this, or did you skip it because we're already validating it above? Maybe we can combine it with the validation above (so only a call to the transformer) and catch the validation error that is throw from the transformer?